user_info.go

```diff
@@ -13,3 +13,20 @@ func (c *WorkwxApp) GetUser(userid string) (*UserInfo, error) {
 	obj := resp.intoUserInfo()
 	return &obj, nil
 }
+
+// GetUserByDept 获取部门成员详情
+func (c *WorkwxApp) GetUserByDeptID(id int64, fetchChild bool) ([]*UserInfo, error) {
+	resp, err := c.execUserList(reqUsersByDeptID{
+		DeptID:     id,
+		FetchChild: fetchChild,
+	})
+	if err != nil {
+		return nil, err
+	}
+	users := make([]*UserInfo, len(resp.users))
+	for _, user := range resp.users {
+		userInfo := user.intoUserInfo()
+		users = append(users, &userInfo)
+	}
```

Comment:
This is wrong, with the result array pre-allocated you must use indexed assignment because the values are already present (with zero values) there.

```go
users := make([]*UserInfo, len(resp.Users))
for i, user := range resp.Users {
    userInfo := user.intoUserInfo()
    users[i] = &userInfo
}
```

With the original code you get `2 * len(resp.Users)` records, first half being all zero values!